ufos/init.lua

```diff
@@ -160,18 +160,15 @@ function ufos.ufo:on_step (dtime)
 		if ctrl.aux1 then
 			local pos = self.object:getpos()
 			local t = {{x=2,z=0},{x=-2,z=0},{x=0,z=2},{x=0,z=-2}}
-			for _, i in ipairs(t) do
-				pos.x = pos.x + i.x; pos.z = pos.z + i.z;
-				if minetest.env:get_node(pos).name == "ufos:furnace" then
-					meta = minetest.env:get_meta(pos)
-					if fuel < 100 and meta:get_int("charge") > 0 then
-						fuel = fuel + 1
-						meta:set_int("charge",meta:get_int("charge")-1)
-						meta:set_string("formspec", ufos.furnace_inactive_formspec
-							.. "label[0,0;Charge: "..meta:get_int("charge"))
-					end
+			furnace_pos = minetest.find_node_near(pos, UFO_CHARGE_RADIUS, "ufos:furnace")
+			if furnace_pos then
```

Comment:
lt looks like this breaks ufo fuelling, it's supposed to work faster with more furnaces.